Dockerfile.prod

```
@@ -0,0 +1,33 @@
+COPY . .
+
+# Install dependencies and build
+RUN pnpm install --frozen-lockfile \
+    && pnpm run build \
+    && pnpm install -P --ignore-scripts --frozen-lockfile
+
+# Stage 2: Production
+FROM node:lts-alpine AS prod
+
+# Install curl for healthcheck
+RUN apk add --no-cache curl
```

Hadolint (2.14.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)